armi/reactor/grids.py

```diff
@@ -952,6 +933,112 @@ def getRingPos(self, indices) -> Tuple[int, int]:
+        return allIndices
+
+    def buildLocations(self):
+        """Populate all grid cells with a characteristic SpatialLocator."""
+        for i, j, k in self.getAllIndices():
+            loc = IndexLocation(i, j, k, self)
+            self._locations[(i, j, k)] = loc
+
+    @property
+    def pitch(self):
+        """
+        The pitch of the grid.
+
+        Assumes 2-d unit-step defined (works for cartesian)
+        """
+        pitch = (self._unitSteps[0][0], self._unitSteps[1][1])
+        if pitch[0] == 0:
+            raise ValueError(f"Grid {self} does not have a defined pitch.")
+        return pitch
+
+
+class CartesianGrid(Grid):
+    """
+    Grid class representing a conformal Cartesian mesh.
+    """
+
+    @classmethod
+    def fromRectangle(
+        cls, width, height, numRings=5, symmetry=None, isOffset=False, armiObject=None
+    ):
+        """
+        Build a finite step-based 2-D Cartesian grid based on a width and height in cm.
+
+        isOffset : bool
+            If True, the origin of the Grid's coordinate system will be placed at the
+            bottom-left corner of the center-most cell. Otherwise, the origin will be
+            placed at the center of the center-most cell.
+
+        """
+        unitSteps = ((width, 0.0, 0.0), (0.0, height, 0.0), (0, 0, 0))
+        offset = numpy.array((width / 2.0, height / 2.0, 0.0)) if isOffset else None
+        return cls(
+            unitSteps=unitSteps,
+            unitStepLimits=((-numRings, numRings), (-numRings, numRings), (0, 1)),
+            offset=offset,
+            armiObject=armiObject,
+            symmetry=symmetry or "",
+        )
+
+    def getRingPos(self, indices):
+        """
+        Return ring and position from indices.
+
+        Ring is the Manhattan distance from (0, 0) to the passed indices. Position
+        counts up around the ring counter-clockwise from the quadrant 1 diagonal, like
+        this::
+
+            7   6  5  4  3  2  1
+            8         |       24
+            9         |       23
+            10 -------|------ 22
+            11        |       21
+            12        |       20
+            13 14 15 16 17 18 19
+
+        Grids that split the central locations have 1 location in in inner-most ring,
+        whereas grids without split central locations will have 4.
+        """
+        i, j = indices[0:2]
+        split = self._isThroughCenter()
+
+        if not split:
+            i += 0.5
+            j += 0.5
+
+        ring = max(abs(int(i)), abs(int(j)))
+
+        if not split:
+            ring += 0.5
+
+        if j == ring:
+            # region 1
+            pos = -i + ring
+        elif i == -ring:
+            # region 2
+            pos = 3 * ring - j
+        elif j == -ring:
+            # region 3
+            pos = 5 * ring + i
+        else:
+            # region 4
+            pos = 7 * ring + j
+        return (int(ring) + 1, int(pos) + 1)
+
+    def locatorInDomain(self, locator):
```

Comment:
docstring for this would look good in sphinx, though if you want to rely o the docstring of the parent class that should be rendered appropriately using default sphinx options

Reply:
sphinx is re-using the base class docstring